### PR TITLE
If file open operations fail, infer and print error messages.

### PR DESCRIPTION
**Pull request by yuezato**
This PR is an ad-hoc patch for the two issues: https://github.com/frugalos/cannyls/issues/9, https://github.com/frugalos/frugalos/issues/36

### CODE REVIEW

File: src/nvm/file.rs [hunk 1]
```diff
@@ -131,7 +215,7 @@ impl FileNvmBuilder {
         // OpenOptions::createはファイルが既に存在する場合はそれを開き
         // 存在しない場合は作成する
         options.create(true);
-        let file = track_io!(options.open(&filepath))?;
+        let file = self.file_open_with_error_info(true, &options, &filepath)?;
```
Comment (sile) [inline src/nvm/file.rs]: Please use `track(...)!` macro at here and other points at where `file_open_with_error_info` method is called.

File: src/nvm/file.rs [hunk 2]
```diff
@@ -115,6 +115,90 @@ impl FileNvmBuilder {
...
+                    file_orig,
+                    format!(
+                        "[CannyLS Error] The file {:?} does not exist and failed to open it.",
+                        filepath.as_ref()
+                    )
+                );
+            }
+        }
+
+        // Next, we check if the file `filepath` can be opened with `O_DIRECT` option.
```
Comment (sile) [inline src/nvm/file.rs]: typo? `s/with/without/`

File: src/nvm/file.rs [hunk 3]
```diff
@@ -115,6 +115,90 @@ impl FileNvmBuilder {
...
+        if file_orig.is_ok() {
+            return file_orig;
+        }
+
+        // Otherwise, we check why opening the file `filepath` failed.
+        // First, we check the existence of the file `filepath`.
+        if !std::path::Path::new(filepath.as_ref()).exists() {
+            if do_create {
+                // failed to file open
+                return track!(
```
Comment (sile) [inline src/nvm/file.rs:139]: This expression can be written as follows:

```rust
return track!(file_orig, "[CannyLS Error] We failed to create the file {:?}.", filepath.as_ref());
```

And I think the header "[CanyLS Error]" is redundant (crate name is included in a `trackable` history).

File: src/nvm/file.rs [hunk 4]
```diff
@@ -115,6 +115,90 @@ impl FileNvmBuilder {
...
+    #[cfg(target_os = "linux")]
+    fn file_open_with_error_info<P: AsRef<Path>>(
+        &self,
+        do_create: bool, // This denoes whether or not the next argument `options` allows file creating.
+        options: &fs::OpenOptions,
+        filepath: &P,
+    ) -> Result<File> {
+        use std::os::unix::fs::OpenOptionsExt;
+
+        let file_orig = track_io!(options.open(&filepath));
```
Comment (sile) [inline src/nvm/file.rs]: I feel "open_result" (or something like "XXX_result") is more appropriate name. How do you think?

Comment (yuezato) [reply src/nvm/file.rs]: OK, I renamed!

**Review by sile — APPROVED** [review]
LGTM

**Comment by shinnya** [comment]
LGTM

**Pull request merged**